src/Symfony/Component/HttpKernel/EventListener/AbstractSessionListener.php

```diff
@@ -115,6 +116,34 @@ public function onKernelResponse(FilterResponseEvent $event)
              * it is saved will just restart it.
              */
             $session->save();
+
+            /*
+             * Write cookie to request object for cli context based runner like roadrunner,
+             * where php can not write the cookie.
+             */
+            if (\PHP_SAPI === 'cli') {
```

Comment:
Maybe we should make this unconditional?

Reply:
Would that not end that in the normal mode (apache php mod, nginx, php-fpm) php itself would write a cookie and symfony would write a cookie with the same name?

Reply:
But maybe we could avoid that at https://github.com/symfony/symfony/blob/9ccd0ad387f0aacf2a1f5673fdcf31dbbef22e35/src/Symfony/Component/HttpFoundation/Response.php#L365 but need to test how it behaves in the normal mode.